File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/exception/ErrorResponse.java
package com.luckyseven.backend.sharedkernel.exception;


import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ErrorResponse {

  private final String code;
  private final String message;
  private final int status;
  private final String detail;

  public static ErrorResponse of(CustomLogicException e) {
    return new ErrorResponse(
        e.getExceptionCode().name(),
        e.getExceptionCode().getMessage(),
        e.getExceptionCode().getHttpStatus().value(),
        e.getDetail()
    );
  }
}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/exception/ErrorResponse.kt
package com.luckyseven.backend.sharedkernel.exception

data class ErrorResponse(
    val code: String,
    val message: String,
    val detail: String?
) {
    companion object {
        fun from(e: CustomLogicException): ErrorResponse {
            return ErrorResponse(
                code = e.exceptionCode.name,
                message = e.exceptionCode.message,
                detail = e.detail
            )
        }
    }
} 